src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/PackageSearch/PackageSearchResultRendererTable.cs

```diff
@@ -0,0 +1,92 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Threading.Tasks;
+using NuGet.Configuration;
+using NuGet.Protocol.Core.Types;
+
+namespace NuGet.CommandLine.XPlat
+{
+    internal class PackageSearchResultRendererTable : IPackageSearchResultRenderer
+    {
+        private PackageSearchArgs _args;
+        private const int LineSeparatorLength = 40;
+        private static readonly string SourceSeparator = new('*', LineSeparatorLength);
+
+        public PackageSearchResultRendererTable(PackageSearchArgs packageSearchArgs)
+        {
+            _args = packageSearchArgs;
+        }
+
+        public async Task Add(PackageSource source, Task<IEnumerable<IPackageSearchMetadata>> completedSearchTask)
+        {
+            _args.Logger.LogMinimal(SourceSeparator);
+
+            if (completedSearchTask == null)
+            {
+                _args.Logger.LogMinimal($"Source: {source.Name} ({source.SourceUri})");
+                _args.Logger.LogMinimal(Strings.Error_CannotObtainSearchSource);
+                return;
+            }
+
+            _args.Logger.LogMinimal($"Source: {source.Name} ({source.SourceUri})");
+            IEnumerable<IPackageSearchMetadata> searchResult = await completedSearchTask;
```

Comment:
The search task can return `null`, right?  That should be handled here so the logic below doesn't cause a null reference exception (NRE).